### ISSUE TITLE
Listener cannot be empty in ServiceProxy for net core

### ISSUE THREAD

**Issue by ThiemeNL** [Azure/service-fabric-issues#1221]
Hi,

Calling a remoting V2_1 listener for a second time causes the service to consume 100% CPU.
The client application is using Microsoft.ServiceFabric.Services.Remoting version 3.2.162 targeting .Net Core 2.1
And the service application runs Microsoft.ServiceFabric.Services.Remoting version 3.2.162 targeting .net 461 and has two listeners V1 and V2_1.

The client code causing the issue:
```C#
   var service = ServiceProxy.Create<IXxService>(
                        new Uri("fabric:/xx/xx"), new ServicePartitionKey(0));
```

And fixing the issue with this: 
```C#
var service = ServiceProxy.Create<IXxService>(
                       new Uri("fabric:/xx/xx"), new ServicePartitionKey(0), 
                       TargetReplicaSelector.Default, "V2_1Listener");
```

After I decompile the SF code, I see that the listener name is only fetched the first time. The second time the this.proxyFactoryV2 is cached, but the listener name keeps empty.
```C#
 public TServiceInterface CreateServiceProxy<TServiceInterface>(Uri serviceUri, ServicePartitionKey partitionKey = null, TargetReplicaSelector targetReplicaSelector = TargetReplicaSelector.Default, string listenerName = null) where TServiceInterface : IService
    {
      Type serviceInterfaceType = typeof (TServiceInterface);
      if (this.proxyFactoryV2 == null)
      {
        ServiceRemotingProviderAttribute providerAttribute = this.GetProviderAttribute(serviceInterfaceType);
        listenerName = this.GetDefaultListenerName(listenerName, providerAttribute.RemotingClientVersion);
        this.proxyFactoryV2 = new Microsoft.ServiceFabric.Services.Remoting.V2.Client.ServiceProxyFactory(new Func<IServiceRemotingCallbackMessageHandler, IServiceRemotingClientFactory>(providerAttribute.CreateServiceRemotingClientFactoryV2), this.retrySettings, (Action<IServiceRemotingClientFactory>) null);
      }
      return this.proxyFactoryV2.CreateServiceProxy<TServiceInterface>(serviceUri, partitionKey, targetReplicaSelector, listenerName);
    }
```

Best regards,
Thieme

**Comment by suchiagicha** [comment]
@ThiemeNL 
Thanks for reporting the issue. We will soon fix this issue .

**Comment by vladislav-karamfilov** [comment]
@suchiagicha , is this now fixed?

**Comment by ThiemeNL** [comment]
@suchiagicha, I'm really worried that this issue is still not closed. Is service fabric dead?

**Comment by raunakpandya** [comment]
@ThiemeNL - Apologies for no responses here. Adding few more team members to comment.

**Comment by amanbha** [comment]
What version of the packages you are using now?

**Comment by vladislav-karamfilov** [comment]
We recently updated to Remoting package version 3.4.641 and didn't observe the problem after that but we thought package update has fixed the problem in the past too and we were wrong... What's the version in which this has been fixed?

**Comment by amanbha** [comment]
@vladislav-karamfilov  this particular issue was fixed with https://github.com/microsoft/service-fabric-services-and-actors-dotnet/pull/143. Fix should be in nupkgs >= 3.3.*. 
In addition to this fix, there were numerous other fixes around service resolution which could cause cpu spikes. These fixes were added over time as discovered with other releases. I se that you are on latest version which should have all the fixes for known issues so far.
